AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -162,24 +162,37 @@ - (void)_layoutNodes:(NSArray *)nodes ofKind:(NSString *)kind atIndexPaths:(NSAr
   }
 
   NSUInteger nodeCount = nodes.count;
-  NSMutableArray<ASCellNode *> *allocatedNodes = [NSMutableArray arrayWithCapacity:nodeCount];
+  NSMutableArray<ASCellNode *> *allocatedNodes = [NSMutableArray<ASCellNode *> arrayWithCapacity:nodeCount];
   dispatch_group_t layoutGroup = dispatch_group_create();
   ASSizeRange *nodeBoundSizes = (ASSizeRange *)malloc(sizeof(ASSizeRange) * nodeCount);
+
   for (NSUInteger j = 0; j < nodes.count && j < indexPaths.count; j += kASDataControllerSizingCountPerProcessor) {
     NSInteger batchCount = MIN(kASDataControllerSizingCountPerProcessor, indexPaths.count - j);
 
+    __block NSArray *subarray;
+    // Allocate nodes concurrently.
     dispatch_block_t allocationBlock = ^{
-      for (NSUInteger k = j; k < j + batchCount; k++) {
+      __strong ASCellNode **allocatedNodeBuffer = (__strong ASCellNode **)calloc(batchCount, sizeof(ASCellNode *));
+      dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+      dispatch_apply(batchCount, queue, ^(size_t i) {
+        unsigned long k = j + i;
```

Comment:
It would be great to make these variable names more descriptive, although I know the original code use these terms, So this is fine for now.